tensorflow/core/kernels/map_kernels.h

```diff
@@ -176,6 +174,39 @@ class TensorMapHasKey : public OpKernel {
   }
 };
 
+class TensorMapStackKeys : public OpKernel {
+ public:
+  explicit TensorMapStackKeys(OpKernelConstruction* c) : OpKernel(c) {
+    OP_REQUIRES_OK(c, c->GetAttr("key_dtype", &key_dtype_));
+  }
+  ~TensorMapStackKeys() override {}
+
+  void Compute(OpKernelContext* c) override {
+    const TensorMap* m = nullptr;
+    OP_REQUIRES_OK(c, GetInputMap(c, 0, &m));
+    
+    OP_REQUIRES(c, m->size() != 0,
+                errors::InvalidArgument("Empty map has no keys."));
```

Comment:
nit: Slightly more informative: "TensorMapStackKeys must not be called on an empty map."